### DISCUSSION TITLE
How can I put some data into RAM before I start the simulation ?

### DISCUSSION THREAD

**Discussion by zq7489** [hneemann/Digital#635] [Q&A]
I use a clock input and a RAM(Chip Select) in my circuit. 
After I start the simulation, I can edit the data in the RAM.
But if I use a real time clock, I will have no time to edit the RAM.
So, how can I edit the data in the RAM before I start the simulation ?
Thanks a lot.

**Comment by hneemann** [comment]
Think of the RAM as a real RAM. How can you store data in a real RAM before it is turned on?
If you want to do something like this, you can use an EEPROM or a ROM.

There is a way to do this in the simulator though, which is mainly intended for debugging processors: you can flag the RAM component as program memory and use the "Circuit specific settings" to load a program into the RAM before starting the simulation.
